source/docs/advanced/theme/wysiwyg.md

```diff
@@ -4,116 +4,385 @@ title: Display WYSIWYG content
+  data: [WysiwygNodeData]
+}
+```
+    3. Register this new type as a Wysiwyg type in your GraphQL module
+```diff
+// my-module/server/modules/wysiwyg/index.js
+contextEnhancer: ({ loaders }) => {
++  loaders.Wysiwyg.registerWysiwygType(
++    // the name of the GraphQL type you have just created
++    "CustomWysiwyg",
++    // The list of shortcodes that are parsed and replaced in your wysiwyg content
++    [
++      {
++        // regex to identify the shortcode within your HTML code
++        regex: /\{\{media url="(.*?)"\}\}/gi,
++        // How to replace the given shortcode
++        replacement: (match, url) => `/media/${url}`
++      }
++    ]
++  );
+}
+```
+2. (Client side) Map this new type to a specific React component that will apply the needed transformations before displaying the content
+    1. Duplicate `theme/modules/WysiwygV2/DefaultWysiwyg` to `theme/modules/WysiwygV2/CustomWysiwyg`.
+    1. Override `theme/modules/WysiwygV2/getWysiwygComponent.js` and map your new `CustomWysiwyg` typename (the one used in your GraphQL schema) to `theme/modules/WysiwygV2/CustomWysiwyg` (the file that tells React how to display the content)
+```diff
+const typenameMap = {
+-  MagentoWysiwyg: loadable(() => import("./MagentoWysiwyg/MagentoWysiwyg"))
++  MagentoWysiwyg: loadable(() => import("./MagentoWysiwyg/MagentoWysiwyg")),
++  CustomWysiwyg: loadable(() => import("./CustomWysiwyg"))
+};
+```
+
+This is it, you have a new `CustomWysiwyg` type. However, the components rendered are still simple HTML tags. If you want to add interactivity, you will need to refer to the `theme/modules/WysiwygV2/CustomWysiwyg` you have created.
+
+In this file you will notice that there is a `defaultComponentsMap` constant. This object is used to reference how to render each html tag in your Wysiwyg component. We call these **transforms**: they transform an html tag into a React component. This is what will allow you to finely tune how your Wysiwyg is displayed and add interactivity to your content.
+
+For instance, if you want your `h2` tags to have some additional classes, you could use the following `defaultComponentsMap` instead:
 
 ```js
-{
-  // regex to identify the shortcode within your HTML code
-  regex: /\{\{media url="(.*?)"\}\}/gi,
-  // How to replace the given shortcode
-  replacement: (match, url) => `/media/${url}`
+const defaultComponentsMap = {
+  h2: (props) => <h2 {...props} className="wysiwyg-h2" />
 }
 ```
 
-See [EnhanceWysiwyg's reference](/docs/reference/wysiwyg.html) for more details.
+The props available represent the HTML attributes coming from your Wysiwyg content. The only difference compared to traditional HTML is that `class` is renamed to `className` to better support React patterns and the `props.children` is an already constructed React element.
+
+<blockquote class="important">
+<b>Note:</b> If the component is complex, please think about code splitting them to avoid too large bundle on pages that don't need them.
```

Comment:
Since code splitting should only be using @loadable/component in the context of FC, I'd say we should wait for the accurate page documentation instead of pointing to a page that would point the readers in the wrong direction.

Reply:
LGTM, and I noticed `@loadable/component` was mentioned later too.